app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt

```diff
@@ -24,12 +25,36 @@ import org.koin.core.KoinComponent
 import org.koin.core.inject
 import timber.log.Timber
 
-class ExternalPlayer(private val fragment: WebViewFragment) : KoinComponent {
-    private val context: Context = fragment.requireContext()
+class ExternalPlayer(
+    private val context: Context,
+    lifecycleOwner: LifecycleOwner,
+    registry: ActivityResultRegistry,
+) : KoinComponent {
 
     private val appPreferences: AppPreferences by inject()
     private val webappFunctionChannel: WebappFunctionChannel by inject()
 
+    private val playerContract = registry.register("externalplayer", lifecycleOwner, ActivityResultContracts.StartActivityForResult()) { result ->
+        val resultCode = result.resultCode
+        val intent = result.data
+        when (val action = intent?.action) {
+            Constants.MPV_PLAYER_RESULT_ACTION -> handleMPVPlayer(resultCode, intent)
+            Constants.MX_PLAYER_RESULT_ACTION -> handleMXPlayer(resultCode, intent)
+            Constants.VLC_PLAYER_RESULT_ACTION -> handleVLCPlayer(resultCode, intent)
+            else -> {
+                if (action != null && resultCode != Activity.RESULT_CANCELED) {
+                    Timber.d("Unknown action $action [resultCode: $resultCode]")
```

Comment:
```suggestion
                    Timber.d("Unknown action $action (resultCode: $resultCode)")
```

Reply:
I actually prefer the square brackets here..